src/MongoDB.AspNetCore.OData/MongoEnableQueryAttribute.cs

```diff
@@ -96,14 +96,14 @@ private IQueryable RewriteExpression(IQueryable queryable)
 
     public override void OnActionExecuted(ActionExecutedContext actionExecutedContext)
     {
-        var originalMongoQueryable = (actionExecutedContext.Result as ObjectResult)?.Value as IMongoQueryable;
+        var originalMongoQueryable = (actionExecutedContext.Result as ObjectResult)?.Value as IQueryable;
         base.OnActionExecuted(actionExecutedContext);
 
         var response = actionExecutedContext.HttpContext.Response;
         // Check whether the response is set and successful
         if (response?.IsSuccessStatusCode() == true && actionExecutedContext.Result is ObjectResult responseContent)
         {
-            if (responseContent.Value is not IMongoQueryable mongoQueryable || originalMongoQueryable != mongoQueryable)
+            if (responseContent.Value is not IQueryable mongoQueryable || originalMongoQueryable != mongoQueryable)
```

Comment:
I suggest to change the variables naming as well. It's not longer `mongoQueryable`?

Reply:
Done.